src/modal/ShareModal.tsx

```diff
@@ -180,10 +182,23 @@ export const NeoShareModal = ({ connection, loadDashboardListFromNeo4j, loadData
                                 }
                             }}
                         />
+                        <NeoSetting key={"selfHosted"} name={"selfHosted"}
+                                    value={selfHosted}
+                                    style={{ marginLeft: "0px", width: "100%", marginBottom: "10px" }}
+                                    type={SELECTION_TYPES.LIST}
+                                    helperText={"Share the dashboard using self Hosted Neodash, otherwise neodash.graphapp.io will be used"}
+                                    label={"Self Hosted Dashboard"}
+                                    defaultValue={"No"}
+                                    choices={["Yes", "No"]}
+                                    onChange={(e) => {
+                                        setShareLink(null);
+                                        setSelfHosted(e);
+                                    }}
+                        />
                         <Button
                             component="label"
                             onClick={(e) => {
-                                setShareLink((shareBaseURL + "/?share&type=" + shareType + "&id=" + encodeURIComponent(shareID) + "&dashboardDatabase=" + encodeURIComponent(dashboardDatabase) +
+                                setShareLink((( selfHosted == "Yes" ? window.location.origin :shareBaseURL ) + "/?share&type=" + shareType + "&id=" + encodeURIComponent(shareID) + "&dashboardDatabase=" + encodeURIComponent(dashboardDatabase) +
```

Comment:
We should disable this feature for Neo4j Desktop. I tried and `window.location.origin` in Desktop resolves to `file:///`, so I suppose that'll generate broken links for users.

I guess the easiest fix would be to disable the button altogether when `window.location.origin` equals `file:///`.

Reply:
Done